packages/crypto/src/crypto.ts

```diff
@@ -536,31 +536,90 @@ const bigIntToHex = (num: bigint, length: number): string => {
 export const fromDerSignature = (derSignature: string): Uint8Array => {
   const derSignatureBuf = uint8ArrayFromHexString(derSignature);
 
-  // Check and skip the sequence tag (0x30)
-  let index = 2;
+  // Check minimum length
+  if (derSignatureBuf.length < 2) {
+    throw new Error(
+      "failed to convert DER-encoded signature: insufficient length",
+    );
+  }
+
+  // Check SEQUENCE tag (0x30 at first byte)
+  if (derSignatureBuf[0] !== 0x30) {
+    throw new Error(
+      "failed to convert DER-encoded signature: invalid format (missing SEQUENCE tag)",
+    );
+  }
+
+  // Check second byte, start of length field
+  let index = 1;
+  const lengthByte = derSignatureBuf[index]!;
+  if (lengthByte <= 0x7f) {
+    // Short form: single byte length
+
+    // directly take the consumed value as length and check against buffer
+    // buffer length: initial header bytes + claimed remaining length
+    if (derSignatureBuf.length < 1 + 1 + lengthByte) {
+      throw new Error(
+        "failed to convert DER-encoded signature: insufficient length",
```

Comment:
I think this message should be updated to be different than the first message? Maybe ("expected X, got Y")...just so there's a difference between passing a < 2 byte value and messing up the length byte encoding.